src/passes/RemoveUnusedModuleElements.cpp

```diff
@@ -199,15 +260,42 @@ struct RemoveUnusedModuleElements : public Pass {
       importsMemory = true;
     }
     // For now, all functions that can be called indirectly are marked as roots.
+    // TODO: Compute this based on which ElementSegments are actually reachable,
+    //       and which functions have a call_indirect of the proper type.
     ElementUtils::iterAllElementFunctionNames(module, [&](Name& name) {
       roots.emplace_back(ModuleElementKind::Function, name);
     });
     // Compute reachability starting from the root set.
     ReachabilityAnalyzer analyzer(module, roots);
+
+    // RefFuncs that are never called are a special case: We cannot remove the
+    // function, since then (ref.func $foo) would not validate. But if we know
```

Comment:
Can't we use `(ref.null func)`?

Reply:
Hmm, I don't think so. The type of a null is nullable while `(ref.func $foo)` is non-nullable. Also, someone might do `ref.is_null` on the ref, even if the ref is never called.